### PR TITLE
Align x5c and thumbprint handling with RFC

### PR DESCRIPTION
**Pull request by ybasket**
According to the [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517#section-4.7), the `x5c` field of a JWK must be non-empty, hence this converts it from `List` to `Option[NonEmptyList]`. This broke unexpectedly the tests for thumbprinting as it didn't align with the [RFC 7638](https://datatracker.ietf.org/doc/html/rfc7638#section-3.2), so the thumbprinting got fixed as well, now taking only required fields into concern as the RFC states.

More comments inline.

As this is a binary breaking change, I suggest to release it as a new major version.

### CODE REVIEW

File: library/src/main/scala/com/chatwork/scala/jwk/ECJWK.scala [hunk 1]
```diff
@@ -62,7 +63,8 @@ object ECJWK extends ECJWKJsonImplicits {
           x509CertificateChain,
           d,
           privateKey,
-          expireAt
+          expireAt,
+          keyStore
```
Comment (ybasket) [inline library/src/main/scala/com/chatwork/scala/jwk/ECJWK.scala:67]: This one is an unrelated bug I found through a compiler warning that `keyStore` was unused. Might make sense to turn on fatal warnings so that these show up as errors.

File: library/src/main/scala/com/chatwork/scala/jwk/JWKThumbprint.scala [hunk 2]
```diff
@@ -11,13 +11,20 @@ import com.github.j5ik2o.base64scala.{ Base64String, Base64StringFactory }
...
+    jwk.keyType match {
+      case KeyType.EC  => computeFromJson(jwk.asJson.mapObject(_.filterKeys(requiredKeysForEC.contains)), hashAlg)
+      case KeyType.RSA => computeFromJson(jwk.asJson.mapObject(_.filterKeys(requiredKeysForRSA.contains)), hashAlg)
+    }
 
   private def getDigest(json: Json, hashAlg: String): Either[JWKThumbprintError, Array[Byte]] = {
     try {
       val md = MessageDigest.getInstance(hashAlg)
-      md.update(json.noSpaces.getBytes(Charset.defaultCharset()))
+      md.update(json.noSpacesSortKeys.getBytes(Charset.defaultCharset()))
```
Comment (ybasket) [inline library/src/main/scala/com/chatwork/scala/jwk/JWKThumbprint.scala:27]: The RFC 7638 also requires the keys to be sorted in lexicographic order.

File: library/src/test/scala/com/chatwork/scala/jwk/ECJWKSpec.scala [hunk 3]
```diff
@@ -90,7 +90,7 @@ class ECJWKSpec extends AnyFreeSpec with Matchers with ECJWKJsonImplicits with E
       jwk.keyType shouldBe KeyType.EC
       jwk.curve shouldBe Curve.P_521
       val result = jwk.computeThumbprint.value
-      result.asString shouldBe "EroitZ-og3Ji6ENuMuey6vEz4hA2i56rOJHfrTeDHII"
+      result.asString shouldBe "rz4Ohmpxg-UOWIWqWKHlOe0bHSjNUFlHW5vwG_M7qYg"
```
Comment (ybasket) [inline library/src/test/scala/com/chatwork/scala/jwk/ECJWKSpec.scala:93]: I verified these with https://kjur.github.io/jsrsasign/sample/tool_jwktp.html, but as they're taken from https://jose.ietf.narkive.com/GnnXSWEA/jwk-thumbprint-rfc-7638 and the thumbprints have been checked there, the confidence in them being correct is even higher.

**Comment by j5ik2o** [comment]
@ybasket
Thank you for your contribution. I plan to incorporate this change. We will release a major version up, so please wait a while.

**Review by j5ik2o — APPROVED** [review]
LGTM!

**Comment by btisdall** [comment]
👍
